pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -2126,6 +2126,18 @@ double Object_getDoubleValue(JSOBJ Py_UNUSED(obj), JSONTypeContext *tc) {
     return GET_TC(tc)->doubleValue;
 }
 
+const char *Object_getBigNumStringValue(JSOBJ obj, JSONTypeContext *tc, 
+                                    size_t *_outLen) {
+    PyObject* repr = PyObject_Repr(obj);
+    PyObject* str = PyUnicode_AsEncodedString(repr, "utf-8", "~E~");
+    char *bytes = PyBytes_AS_STRING(str);
+
+    PyObject_Free(GET_TC(tc)->cStr);
+    GET_TC(tc)->cStr = bytes;
```

Comment:
So to manage the state properly you should allocate storage for bytes and memcpy whatever PyUnicode_AsUTF8AndSize returns to it. You will need to then free that memory in the callback function for ObjectEnd